packages/tailwindcss/src/variants.ts

```diff
@@ -933,19 +1045,25 @@ export function createVariants(theme: Theme): Variants {
     }
   }
 
-  staticVariant('portrait', ['@media (orientation: portrait)'], { compounds: false })
-  staticVariant('landscape', ['@media (orientation: landscape)'], { compounds: false })
+  staticVariant('portrait', ['@media (orientation: portrait)'], { compounds: ['at-rule'] })
```

Comment:
Since `staticVariant` automatically sets the `compounds` array for style rules, I would expect it to also work automatically for at-rules. Can we make it so that the `staticVariant` function checks the rules and if one starts with `@` it will set this automatically?